worlds/oot_soh/location_access/overworld/market.py

```diff
@@ -5,8 +5,273 @@
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT4, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT5, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT6, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT7, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT8, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT9, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT10, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_ADULT_POT11, lambda bundle: (is_adult(bundle) and can_break_pots(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE1, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE2, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE3, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE4, lambda bundle: (is_child(bundle) and can_break_crates(bundle))),
+        (Locations.MARKET_GUARD_HOUSE_CRATE5, lambda bundle: (is_child(bundle) and can_break_crates(bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_GUARD_HOUSE, world, [
+        (Regions.MARKET_ENTRANCE, lambda bundle: True)
+    ])
+    
+    ## Market Bazaar
+    # Locations
+    add_locations(Regions.MARKET_BAZAAR, world, [
+        (Locations.MARKET_BAZAAR_ITEM1, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM2, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM3, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM4, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM5, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM6, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM7, lambda bundle: True),
+        (Locations.MARKET_BAZAAR_ITEM8, lambda bundle: True),
+    ])
+    # Connections
+    connect_regions(Regions.MARKET_BAZAAR, world, [
+        (Regions.MARKET, lambda bundle: True)
+    ])
+    
+    ## Market Mask Shop
+    # Events
+    add_events(Regions.MARKET_MASK_SHOP, world, [
+        (EventLocations.MARKET_MASK_SHOP_ALL_MASKS, Events.CAN_BORROW_MASKS, lambda bundle: (has_item(Items.ZELDAS_LETTER, bundle) and
+                                                                                   (has_item(Events.KAKARIKO_GATE_OPEN, bundle) or world.options.kakariko_gate.value == 1) or
+                                                                                   world.options.complete_mask_quest.value == 1)),
```

Comment:
It looks like ship assumes physical access to the gate even with the gate set to open from the start. This is on me, but I think we should just revert this to follow ship directly. So essentially set the requirement to having zelda's letter and having the kak gate event.

Also learned that having the gate set to open allows mask shuffling, I was about to comment on that part until I saw Ship does the same with a comment explaining the bug lol.